Event subscription tests has random failures

e.g. Running tests -

```
 1615 passing (13s)
  6 pending
  1 failing

  1) util/subscribe-to-events checks for new events regularly:
     AssertionError: expected 0 to be at least 1
      at Context._callee5$ (src/util/subscribe-to-events.spec.js:113:38)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:64:40)
      at GeneratorFunctionPrototype.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:355:22)
      at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:116:21)
      at step (node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
      at run (node_modules/core-js/library/modules/es6.promise.js:87:22)
      at node_modules/core-js/library/modules/es6.promise.js:100:28
      at flush (node_modules/core-js/library/modules/_microtask.js:18:9)
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at process._tickCallback (internal/process/next_tick.js:98:9)
```

Re-running the tests -

```
1616 passing (13s)
  6 pending
```

More today & last night -

```
 1617 passing (17s)
  6 pending
  1 failing

  1) util/subscribe-to-events checks for new events regularly:
     AssertionError: expected 0 to be at least 1
      at Context._callee5$ (src/util/subscribe-to-events.spec.js:113:38)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:64:40)
      at GeneratorFunctionPrototype.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:355:22)
      at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:116:21)
      at step (node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
      at run (node_modules/core-js/library/modules/es6.promise.js:87:22)
      at node_modules/core-js/library/modules/es6.promise.js:100:28
      at flush (node_modules/core-js/library/modules/_microtask.js:18:9)
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at process._tickCallback (internal/process/next_tick.js:98:9)
```

--- a/js/src/util/subscribe-to-events.js
+++ b/js/src/util/subscribe-to-events.js
@@ -76,7 +76,8 @@ const subscribeToEvents = (contract, events, opt = {}) => {
 
   filter = api.eth
     .newFilter({
-      fromBlock: opt.from, toBlock: opt.to,
+      fromBlock: opt.from,
+      toBlock: opt.to,
       address: contract.address,
       topics: [signatures]
     })

--- a/js/src/util/subscribe-to-events.spec.js
+++ b/js/src/util/subscribe-to-events.spec.js
@@ -14,15 +14,18 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-import { spy, stub } from 'sinon';
+import { spy, stub, useFakeTimers } from 'sinon';
 
 import subscribeToEvents from './subscribe-to-events';
 import {
   pastLogs, liveLogs, createApi, createContract
 } from './subscribe-to-events.test.js';
 
+// Note: We want to have a `setTimeout` that is independent from
+// `sinon.useFakeTimers`. Therefore we dereference `setTimeout` here.
+const _setTimeout = setTimeout;
 const delay = (t) => new Promise((resolve) => {
-  setTimeout(resolve, t);
+  _setTimeout(resolve, t);
 });
 
 describe('util/subscribe-to-events', () => {
@@ -97,7 +100,8 @@ describe('util/subscribe-to-events', () => {
     expect(api.eth.uninstallFilter.firstCall.args).to.eql([ 123 ]);
   });
 
-  it.skip('checks for new events regularly', async function () {
+  it('checks for new events regularly', async function () {
+    const clock = useFakeTimers();
     const { api, contract } = this;
 
     api.eth.getFilterLogs = stub().resolves([]);
@@ -108,12 +112,16 @@ describe('util/subscribe-to-events', () => {
     subscribeToEvents(contract, [ 'Bar' ], { interval: 5 })
       .on('log', onLog)
       .on('Bar', onBar);
-    await delay(10);
+    await delay(1); // let stubs resolve
+    clock.tick(5);
+    await delay(1); // let stubs resolve
 
     expect(onLog.callCount).to.be.at.least(1);
     expect(onLog.firstCall.args).to.eql([ liveLogs[0] ]);
     expect(onBar.callCount).to.be.at.least(1);
     expect(onBar.firstCall.args).to.eql([ liveLogs[0] ]);
+
+    clock.restore();
   });
 
   it('accepts a custom block range', async function () {

